.NET/Microsoft.Recognizers.Text.DateTime/Parsers/BaseDatePeriodParser.cs

```diff
@@ -1873,15 +1873,13 @@ private DateTimeResolutionResult ParseQuarter(string text, DateObject referenceD
                 {
                     ret.FutureValue = ret.PastValue = new Tuple<DateObject, DateObject>(beginDate, endDate);
                 }
-
-                ret.Timex = $"({DateTimeFormatUtil.LuisDate(-1, beginDate.Month, 1)},{DateTimeFormatUtil.LuisDate(-1, endDate.Month, 1)},P3M)";
             }
             else
             {
                 ret.FutureValue = ret.PastValue = new Tuple<DateObject, DateObject>(beginDate, endDate);
-                ret.Timex = $"({DateTimeFormatUtil.LuisDate(beginDate)},{DateTimeFormatUtil.LuisDate(endDate)},P3M)";
             }
 
+            ret.Timex = TimexUtility.GenerateDatePeriodTimex(beginDate, endDate, DatePeriodTimexType.ByMonth, noSpecificYear);
```

Comment:
The last parameter should not be a bool. It needs to be able to represent any part of the missing context, not only year.
Maybe a flags enum?

Reply:
What about UnspecificDateTimeTerm?